src/io/free_functions.rs

```diff
@@ -235,8 +235,7 @@ pub(crate) fn guess_format_from_path_impl(path: &Path) -> Result<ImageFormat, Pa
         .and_then(|s| s.to_str())
         .map(str::to_ascii_lowercase);
 
-    let ext = ext.as_ref()
-        .map(String::as_str);
+    let ext = ext.as_deref();
```

Comment:
I think this requires rust 1.40.0, while `image` still promises backwards compatibility to 1.36.0.

Reply:
With `1.34.2` to be exact.

Reply:
Thank you! I've reverted this.